cmd/admin-handlers.go

```diff
@@ -1372,6 +1373,135 @@ func (a adminAPIHandlers) ServerHardwareInfoHandler(w http.ResponseWriter, r *ht
 	}
 }
 
+// OBDInfoHandler - GET /minio/admin/v2/obdinfo
+// ----------
+// Get server on-board diagnostics
+func (a adminAPIHandlers) OBDInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "OBDInfo")
+	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.OBDInfoAdminAction)
+	if objectAPI == nil {
+		return
+	}
+	deadlinedCtx, cancel := context.WithDeadline(ctx, time.Now().Add(10*time.Minute))
+	obdInfo := madmin.OBDInfo{}
+
+	setCommonHeaders(w)
+	w.Header().Set(xhttp.ContentType, string(mimeJSON))
+	w.WriteHeader(http.StatusOK)
+
+	partialWrite := func() {
+		jsonBytes, _ := json.MarshalIndent(obdInfo, " ", "  ")
+		_, err := w.Write(jsonBytes)
+		logger.LogIf(ctx, err)
+	}
+
+	finish := func() {
+		partialWrite()
+		w.(http.Flusher).Flush()
+		cancel()
+	}
+
+	errResp := func(err error) {
+		errorResponse := getAPIErrorResponse(ctx, toAdminAPIErr(ctx, err), r.URL.String(),
+			w.Header().Get(xhttp.AmzRequestID), globalDeploymentID)
+		encodedErrorResponse := encodeResponse(errorResponse)
+		obdInfo.Error = string(encodedErrorResponse)
+		finish()
+	}
+
+	// returns a locked lock
+	lock, err := objectAPI.GetRWLock(deadlinedCtx, "obd", 5*time.Second, 2*time.Second)
+	if err != nil {
+		errResp(err)
+		return
+	}
+	defer lock.Unlock()
+	vars := mux.Vars(r)
+
+	if cpu, ok := vars["syscpu"]; ok && cpu == "true" {
+		cpuInfo := getLocalCPUOBDInfo(deadlinedCtx)
```

Comment:
getLocalCPUOBDInfo() can take the `r.Host` to make sure that when it's in FS or XL mode not in distributed you don't default to `minio` as the hostname.

Reply:
Look at getLocalMemUsage()

Reply:
This needs to be repeated across all functions

Reply:
That;s what I started with. However, you end up getting the path as well, when using `r.Host`

Reply:
why would you get the path? @wlan0 not able to understand?

Reply:
I thought you said `r.Host`, not `r.Hostname` my bad